python/paddle/fluid/dygraph/dygraph_to_static/utils.py

```diff
@@ -1195,3 +1196,39 @@ def _is_wrapped(f):
         unwrapped_f = unwrapped_f.__wrapped__
 
     return unwrapped_f
+
+
+def input_specs_compatible(src_input_specs, other_input_specs):
+    """
+    Returns True if the two input specs are compatible, otherwise False.
+
+    args:
+        src_input_spec (list[InputSpec]|tuple(InputSpec)): list/tuple of
+            paddle.static.InputSpec
+        other_input_spec (list[InputSpec]|tuple(InputSpec)): list/tuple of
+            paddle.static.InputSpec
+    """
+    len_specs = len(src_input_specs)
+    if len_specs != len(other_input_specs):
+        return False
+
+    for i in range(len_specs):
+        src_shape = src_input_specs[i].shape
+        other_shape = other_input_specs[i].shape
```

Comment:
`src_input_specs[i]` may be not a `InputSpec` but a list or dict, etc.  `src_input_specs[i].shape` may raise an exception.  Better to add  code like  `if isinstance(rc_input_specs[i], InputSpec)` to make sure it's type is `InputSpec`

Reply:
1. No, in this case users can debug the input type is incorrect. By the way, if we add something like `assert isinstance(src_input_specs[i], InputSpec)`, I have to import InputSpec at utils.py which can cause recursive import error. This may due to our current Python import dependency organization is not very good. We may change it in the future.

You can see now we have 1200+ lines in `utils.py`, this is a large utils file now. We may consider to split it in the future and handle the import dependency better.

2. No, the name doesn't check. Our input spec checks arg names when it is `dict` whose keys are arg names.